analytical_engine/apps/assortativity/degree/degree_assortativity_coefficient.h

```diff
@@ -0,0 +1,218 @@
+template <typename FRAG_T>
+class DegreeAssortativity
+    : public AppBase<FRAG_T, DegreeAssortativityContext<FRAG_T>>,
+      public grape::Communicator {
+ public:
+  INSTALL_DEFAULT_WORKER(DegreeAssortativity<FRAG_T>,
+                         DegreeAssortativityContext<FRAG_T>, FRAG_T)
+  static constexpr grape::MessageStrategy message_strategy =
+      grape::MessageStrategy::kAlongEdgeToOuterVertex;
+  static constexpr grape::LoadStrategy load_strategy =
+      grape::LoadStrategy::kBothOutIn;
+  using vertex_t = typename fragment_t::vertex_t;
+  using vid_t = typename fragment_t::vid_t;
+  using oid_t = typename fragment_t::oid_t;
+  using edata_t = typename fragment_t::edata_t;
+  using degree_t = double;
+
+  void PEval(const fragment_t& frag, context_t& ctx,
+             message_manager_t& messages) {
+    auto inner_vertices = frag.InnerVertices();
+    for (auto v : inner_vertices) {
+      ProcessVertex(v, frag, ctx, messages);
+    }
+    // if workers_num = 1, force continue
+    messages.ForceContinue();
+  }
+
+  void IncEval(const fragment_t& frag, context_t& ctx,
+               message_manager_t& messages) {
+    if (!ctx.merge_stage) {
+      vertex_t v;
+      degree_t source_degree;
+      while (messages.GetMessage(frag, v, source_degree)) {
+        degree_t target_degree =
+            GetDegreeByType(frag, v, ctx.target_degree_type_, ctx);
+        DegreeMixingCount(source_degree, target_degree, ctx);
+      }
+      ctx.merge_stage = true;
+      if (frag.fid() != 0) {
+        messages.SendToFragment(0, ctx.degree_mixing_map);
+      }
+      messages.ForceContinue();
+    } else {
+      // merge in work 0
+      if (frag.fid() == 0) {
+        // {{source_degree, {target_degree, num}}
+        std::unordered_map<degree_t, std::unordered_map<degree_t, int>> msg;
+        while (messages.GetMessage(msg)) {
+          for (auto& pair1 : msg) {
+            for (auto& pair2 : pair1.second) {
+              // merge
+              if (ctx.degree_mixing_map.count(pair1.first) == 0 ||
+                  ctx.degree_mixing_map[pair1.first].count(pair2.first) == 0) {
+                ctx.degree_mixing_map[pair1.first][pair2.first] = pair2.second;
+              } else {
+                ctx.degree_mixing_map[pair1.first][pair2.first] += pair2.second;
+              }
+            }
+          }
+        }
+        std::vector<std::vector<double>> degree_mixing_matrix;
+        std::unordered_map<int, degree_t> map;
+        // get degree mixing matrix
+        GetDegreeMixingMatrix(ctx, degree_mixing_matrix, map);
+        ctx.degree_assortativity = ProcessMatrix(degree_mixing_matrix, map);
+
+        std::vector<size_t> shape{1};
+        ctx.set_shape(shape);
+        ctx.assign(ctx.degree_assortativity);
+        VLOG(0) << "degree assortatity: " << ctx.degree_assortativity
+                << std::endl;
+      }
+    }
+  }
+
+ private:
+  void ProcessVertex(const vertex_t& v, const fragment_t& frag, context_t& ctx,
```

Comment:
add some comment for the function and use a more reasonable name, if is private method, we should lowercase first letter, FYI https://zh-google-styleguide.readthedocs.io/en/latest/google-cpp-styleguide/